addons/account/models/account_tax_report.py

```diff
@@ -55,8 +55,8 @@ def copy(self, default=None):
         lines_map = {} # maps original lines to their copies (using ids)
         lines_to_treat = list(self.line_ids.filtered(lambda x: not x.parent_id))
         while lines_to_treat:
-            line = lines_to_treat.pop()
-            lines_to_treat += list(line.children_line_ids)
+            line = lines_to_treat[0]
+            lines_to_treat = list(line.children_line_ids) + lines_to_treat[1:]
```

Comment:
Maybe my brains are not functioning correctly anymore, but isn't this the same?
```suggestion
        for lines_to_treat in self.get_lines_in_hierarchy():

Reply:
It definitely is: I even copy pasted from it to make the change :man_facepalming: 
Thanks